### DISCUSSION TITLE
How do I make the `RPROMPT` align with the top of the `PROMPT`?

### DISCUSSION THREAD

**Discussion by eamonburns** [ohmyzsh/ohmyzsh#11994] [Help] [answered]
I am trying to customize my prompt, and I would like to have an `RPROMPT` I know how to do this, but it isn't quite the right way I would like it to be:

Let's say my prompt is like this:
```
PROMPT=$'%n@%m %~\n> "
RPROMPT='%T'
```
This results in:
```
| agente11@my_laptop ~                |
| >                              9:30 |
|                                     |
```

This is pretty good, but, how would I make the `RPROMPT` align with the top of the `PROMPT`? Like this:
```
| agente11@my_laptop ~           9:30 |
| >                                   |
|                                     |
```
That way, the `RPROMPT` will not be overwritten by a very long command (and, it looks nicer)

**Comment by mcornella** [comment, marked as answer]
You would need to add the `$RPROMPT` contents to your `$PROMPT` on the top line, and add a spacer string (of empty spaces) so that the content is displayed on the right. As far as I know there's no other way around that:

```
PROMPT="%n@%m %~<a bunch of spaces>%T"$'\n> '
```

The spacer can be computed dynamically from the content you display and the amount of characters per row. You can check this very thorough answer: https://unix.stackexchange.com/a/250457

**Reply by eamonburns** [reply]
Ok, thanks. For anyone else who wants to know, what I _really_ wanted was the [powerlevel10k](https://github.com/romkatv/powerlevel10k) theme (I was trying to build it from scratch basically, and it is way easier to use it out of the box). It might not work for everyone, but it might work for you.
